File: android/app/src/main/kotlin/dev/alek6dev/chessclock/ui/theme/Textures.kt
package dev.alek6dev.chessclock.ui.theme

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * Réglure horizontale — seule texture des maquettes officielles : filets fins réguliers,
 * espacement et épaisseur identiques sur tous les écrans (0.5 pt / 5 pt), seule la couleur
 * et l'opacité des filets changent d'un écran à l'autre.
 */
@Composable
fun RuledBackground(
    baseColor: Color,
    lineColor: Color,
    spacing: Dp = 5.dp,
    lineThickness: Dp = 0.5.dp,
    modifier: Modifier = Modifier,
) {
    Canvas(modifier = modifier.fillMaxSize().background(baseColor)) {
        val step = spacing.toPx()
        val thickness = lineThickness.toPx()
        var y = 0f
        while (y < size.height) {
            drawLine(
                color = lineColor,
                start = Offset(0f, y),
                end = Offset(size.width, y),
                strokeWidth = thickness,
            )
            y += step
        }
    }
}

/** Unique tache d'encre du papier vieilli — jamais plus d'une par surface, en haut à droite. */
@Composable
fun InkStainOverlay(modifier: Modifier = Modifier, color: Color = Color(0x148C3B2E)) {
    Canvas(modifier = modifier.fillMaxSize()) {
        val center = Offset(size.width * 0.74f, size.height * 0.12f)
        val radius = size.minDimension * 0.6f
        drawCircle(
            brush = Brush.radialGradient(
                colors = listOf(color, Color.Transparent),
                center = center,
                radius = radius,
            ),
            radius = radius,
            center = center,
        )
    }
}

File: ios/ChessClock/Theme/Textures.swift
import SwiftUI

/// Réglure horizontale — seule texture des maquettes officielles : filets fins réguliers,
/// espacement et épaisseur identiques sur tous les écrans (0.5 pt / 5 pt), seule la couleur
/// et l'opacité des filets changent d'un écran à l'autre.
struct RuledBackground: View {
    var baseColor: Color
    var lineColor: Color
    var spacing: CGFloat = 5
    var lineThickness: CGFloat = 0.5

    var body: some View {
        Canvas { context, size in
            var y: CGFloat = 0
            while y < size.height {
                var path = Path()
                path.move(to: CGPoint(x: 0, y: y))
                path.addLine(to: CGPoint(x: size.width, y: y))
                context.stroke(path, with: .color(lineColor), lineWidth: lineThickness)
                y += spacing
            }
        }
        .background(baseColor)
    }
}

/// Unique tache d'encre du papier vieilli — jamais plus d'une par surface, en haut à droite.
struct InkStainOverlay: View {
    var color: Color = ChessClockColors.lacquerRed.opacity(0.08)

    var body: some View {
        Canvas { context, size in
            let center = CGPoint(x: size.width * 0.74, y: size.height * 0.12)
            let radius = min(size.width, size.height) * 0.6
            let rect = CGRect(x: center.x - radius, y: center.y - radius, width: radius * 2, height: radius * 2)
            context.fill(
                Path(ellipseIn: rect),
                with: .radialGradient(
                    Gradient(colors: [color, .clear]),
                    center: center,
                    startRadius: 0,
                    endRadius: radius
                )
            )
        }
        .allowsHitTesting(false)
    }
}
